test/extended/csrapprover/csrapprover.go

```diff
@@ -0,0 +1,163 @@
+package csrapprover
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	certv1beta1 "k8s.io/api/certificates/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/client-go/kubernetes/scheme"
+	certclientv1beta1 "k8s.io/client-go/kubernetes/typed/certificates/v1beta1"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+func init() {
+	certv1beta1.AddToScheme(scheme.Scheme)
+}
+
+var _ = g.Describe("don't pwn cluster via MCS and autoapprover:", func() {
+	// since login mutates the current kubeconfig we want to use NewCLI
+	// as that will give each one of our test runs a new config via SetupProject
+	oc := exutil.NewCLI("cluster-client-cert", exutil.KubeConfigPath())
+	defer g.GinkgoRecover()
+
+	g.It("deny pod's access to /config/master API endpoint", func() {
+		// the /config/master API port+endpoint is only visible from inside the cluster
+		// (-> we need to create a pod to try to reach it)  and contains the token
+		// of the node-bootstrapper, so no random pods should be able to see it
+		pod, err := exutil.NewPodExecutor(oc, "get-bootstrap-creds", "fedora:latest")
```

Comment:
Minor but let's not use the Docker hub.  I.e. use `registry.fedoraproject.org/fedora:30` *or* use RHEL UBI.